features/cellular/framework/targets/RiotMicro/AT/RM1000_AT_CellularStack.cpp

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright (c) 2018, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "RM1000_AT_CellularStack.h"
+
+#include "mbed_poll.h"
+
+#include "mbed-trace/mbed_trace.h"
+#ifndef TRACE_GROUP
+#define TRACE_GROUP  "RIOT"
+#endif // TRACE_GROUP
+
+using namespace mbed;
+using namespace mbed_cellular_util;
+
+RM1000_AT_CellularStack::RM1000_AT_CellularStack(ATHandler &atHandler, int cid, nsapi_ip_stack_t stack_type) : AT_CellularStack(atHandler, cid, stack_type)
+{
+    tr_debug("RM1000_AT_CellularStack::RM1000_AT_CellularStack");
+
+    // URC handlers for sockets
+    _at.set_urc_handler("+RUSORCV:", callback(this, &RM1000_AT_CellularStack::RUSORCV_URC));
+    _at.set_urc_handler("+RUSOCL:", callback(this, &RM1000_AT_CellularStack::RUSOCL_URC));
+}
+
+RM1000_AT_CellularStack::~RM1000_AT_CellularStack()
+{
+    tr_debug("RM1000_AT_CellularStack::~RM1000_AT_CellularStack");
+
+    _at.set_urc_handler("+RUSORCV:", NULL);
+    _at.set_urc_handler("+RUSOCL:", NULL);
+}
+
+nsapi_error_t RM1000_AT_CellularStack::socket_listen(nsapi_socket_t handle, int backlog)
+{
+    tr_debug("RM1000_AT_CellularStack::socket_listen");
+    return NSAPI_ERROR_UNSUPPORTED;
+}
+
+nsapi_error_t RM1000_AT_CellularStack::socket_accept(void *server, void **socket, SocketAddress *addr)
+{
+    tr_debug("RM1000_AT_CellularStack::socket_accept");
+    return NSAPI_ERROR_UNSUPPORTED;
+}
+
+// Callback for Socket Receive URC.
+void RM1000_AT_CellularStack::RUSORCV_URC()
+{
+    int a, b;
+    CellularSocket *socket;
+
+    a = _at.read_int();
+    b = _at.read_int();
+    tr_debug("RUSORCV_URC called %d %d", a, b);
```

Comment:
See comment below, try to avoid debug print in URC handlers.

Reply:
it was used for debugging. I removed it now